### PR TITLE
Make sure the catalog is updated after setting fields or actions.

### PR DESCRIPTION
**Pull request by mauritsvanrees**
Fixes https://github.com/collective/collective.easyform/issues/8.

### CODE REVIEW

**Comment by mauritsvanrees** [comment]
Travis has the same failure as master, so we could choose to merge this.

Otherwise, after Travis is fixed (likely with PR #247) I can rebase.

File: src/collective/easyform/api.py [hunk 1]
```diff
@@ -158,7 +158,11 @@ def set_fields(context, schema):
     snew_schema = serializeSchema(schema)
     # store the current schema
     context.fields_model = snew_schema
+    # Plain reindexObject() will call notifyModified and then reindex all.
+    # We know nothing has changed that needs indexing,
+    # except that we must update the modified date, and reindex its index.
     context.notifyModified()
+    context.reindexObject(idxs=["modified"])
```
Comment (fredvd) [inline src/collective/easyform/api.py:165]: LGTM, but I didn't  know about .notifyModified() until now.  Is the issue that when you add the idxs parameter, reindexObject does not call notifyModified? And what is is exactly doing, triggering the event which cachepurging also might listen to?

**Review by fredvd — APPROVED** [review]
Lets see if #247 fixes the ci.  I didn't know about notifyModified, don't undersstand the comment fully, but that's more curiosity from my side.

**Comment by mauritsvanrees** [comment]
Other PR merged to master, and I have merged master here. Should fix Travis.

File: src/collective/easyform/api.py (hunk 1 above)
Comment (mauritsvanrees) [reply src/collective/easyform/api.py:165]: I will answer the question about notifyModified, but first I will rebase and force push.

Comment (mauritsvanrees) [reply src/collective/easyform/api.py:165]: I wondered if the comments would disappear after a forced push, which they don't.
Let me answer the question in the main PR.

**Comment by mauritsvanrees** [comment]
So.. you have [`reindexObject`](https://github.com/zopefoundation/Products.CMFCore/blob/2.4.8/Products/CMFCore/CMFCatalogAware.py#L81-L95):

    def reindexObject(self, idxs=[], update_metadata=1, uid=None):

If `idxs` is empty, the default, it calls `notifyModified`, which I will come to.
After that, it reindexes the object. When `idxs` is specified, the object is only reindexed for those indexes. That saves time, especially for `SearchableText`.
It also updates the metadata, which should mean that when you next get the catalog brain for this item, it is totally up to date.

[`notifyModified`](https://github.com/plone/Products.CMFPlone/blob/5.2.2/Products/CMFPlone/DublinCore.py#L95-L99) adds the current user to the list of Creators, but that is beside the point here. The important thing here is that it calls `setModificationDate()` on the object. This is called without an explicit date, so you get a modification date of 'now'.

So what has changed, and why does it work?

Originally, only `notifyModified` was called. So the modification date of the EasyForm object was updated. This *might* have had an effect on caching, but it depends on how caching is setup, and in this case it did not work.

Now I do an extra call: `context.reindexObject(idxs=["modified"])`. I could have actually also called `context.reindexObject()` and it would have had the same effect, just taking slightly longer.
The important effect is **not** that a new modification date is in the index or on the brain.

The important effect could be called a side effect: the **catalog counter** is updated.

This means that the [`catalogCounter` ETag](https://github.com/plone/plone.app.caching/blob/2.0.6/plone/app/caching/operations/etags.py#L121-L138) is changed. So any caching rule that uses this ETag, will conclude that there has been a change, so it no longer returns a '304 Not Modified', but it will return a new response.

That is the important change of this PR.

**Pull request merged**